DemoWebAPI/Controllers/QuizResponseController.cs

```
@@ -14,29 +14,37 @@ public QuizResponseController(DemoContext dc)
         this.dc = dc;
     }
 
+    public class QuestionAnswerPair
+    {
+        public int question { get; set; }
+        public int option { get; set; }
+    }
+
     /// <summary>
     /// This method should add a 'Result' from someone taking the quiz,
     /// along with each of the answers they gave to whichever questions they were asked.
     /// </summary>
     [HttpPut]
-    public IActionResult AddQuizResponse(string userName,int qI1, int o1, int qI2, int o2, int qI3, int o3, int qI4, int o4)
+    public IActionResult AddQuizResponse(string userName, List<QuestionAnswerPair> qAndOList)
     {
-        var x=new QuizResponse {UserName=userName};
+        var x = new QuizResponse
+        {
+            UserName = userName
+        };
 
         dc.QuizResponses.Add(x);
         dc.SaveChanges();
 
-        var a=new QuestionAnswer{QuestionId=qI1,QuestionOptionId=o1,QuizResponseId=x.Id};
-        dc.QuestionAnswers.Add(a);
-
-        var b=new QuestionAnswer{QuestionId=qI2,QuestionOptionId=o2,QuizResponseId=x.Id};
-        dc.QuestionAnswers.Add(b);
-
-        var c=new QuestionAnswer{QuestionId=qI3,QuestionOptionId=o3,QuizResponseId=x.Id};
-        dc.QuestionAnswers.Add(c);
-
-        var d=new QuestionAnswer{QuestionId=qI4,QuestionOptionId=o4,QuizResponseId=x.Id};
-        dc.QuestionAnswers.Add(d);
+        qAndOList.ForEach(q =>
+        {
+            var a = new QuestionAnswer
+            {
+                QuestionId = q.question,
+                QuestionOptionId = q.option,
+                QuizResponseId = x.Id
+            };
+            dc.QuestionAnswers.Add(a);
+        });
 
         dc.SaveChanges();

```

There are two updates here, so it could save the Quiz Response, and then fail on the Question answers and then we'd incorrect data... there should probably also be some error handling so that we can return a nice error instead of a "500 Internal Server Error"